actions/create-pr/src/create-commit.ts

```diff
@@ -0,0 +1,35 @@
+import * as core from '@actions/core';
+import {
+  generateRandomSuffix,
+  gitAdd,
+  gitCheckoutBranch,
+  gitCommit,
+  gitHasChanges,
+} from '@dfinity/action-utils';
+
+export interface CreateCommitOptions {
+  message: string;
+  head: string;
+  authorName: string;
+  authorEmail: string;
+}
+
+export function createCommit({
+  message,
+  head,
+  authorName,
+  authorEmail,
+}: CreateCommitOptions): void {
+  if (!gitHasChanges()) {
+    throw new Error('No changes to commit');
```

Comment:
maybe there should be an option for the client to ignore if there's no changes instead of failling 🤔 

e.g. if we are automating the PR request creation in the CI with commitizen then there would only be changes to the changelog if there are commit types that are relevant for that, in case there's no, there would be no changes, but with this approach of throwing an error we would fail the CI workflow

Reply:
Good point, I made this an early return and added an output `pull_request_created` to be used as a conditional for follow-up steps.